### PR TITLE
PID: Add table for HMPID/RICH

### PR DESCRIPTION
**Pull request by njacazio**

### CODE REVIEW

**Comment by jgrosseo** [comment]
Could we put all ALICE3 stuff in a separate library (still in the Analysis folder of course)?

**Comment by njacazio** [comment]
Indeed, that's a nice idea, also for future development. I created a new directory for the ALICE3 stuff in `Analysis` and I also moved there the analysis task that was in `Tasks`

File: Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx [hunk 1]
```diff
@@ -107,6 +107,8 @@ std::vector<std::string> getColumnNames(header::DataHeader dh)
       return columnNamesTrait(typename StoredTracksCovMetadata::table_t::persistent_columns_t{});
     } else if (description == "TRACK:EXTRA") {
       return columnNamesTrait(typename TracksExtraMetadata::table_t::persistent_columns_t{});
+    } else if (description == "HMPID") {
+      return columnNamesTrait(typename TracksHMPID::table_t::persistent_columns_t{});
```
Comment (jgrosseo) [inline Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx]: This is not needed (it is only needed for the tracks table)

Comment (njacazio) [reply Framework/AnalysisSupport/src/AODJAlienReaderHelpers.cxx]: I moved the HMPID table and renamed `HMPIDs`

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 2]
```diff
@@ -276,6 +276,21 @@ DECLARE_SOA_TABLE(UnassignedTracks, "AOD", "UNASSIGNEDTRACK",
...
+DECLARE_SOA_COLUMN(HMPIDDistance, hmpidDistance, float);
+DECLARE_SOA_COLUMN(HMPIDQMip, hmpidQMip, short);
+} // namespace hmpid
+
+DECLARE_SOA_TABLE(HMPIDs, "AOD", "HMPID",
+                  hmpid::TrackId,
+                  hmpid::HMPIDSignal,
+                  hmpid::HMPIDDistance,
+                  hmpid::HMPIDQMip);
+
```
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h:297]: Last thing
Add `using HMPID = HMPIDs::iterator;`

Comment (njacazio) [reply Framework/Core/include/Framework/AnalysisDataModel.h:297]: Hi, sure, I'm also investigating why I cannot get it to compile in the CI

**Comment by jgrosseo** [comment]
It still fails...

**Comment by njacazio** [comment]
Indeed, shall I convert it to draft?

**Comment by jgrosseo** [comment]
You cannot fix the compilation issue?

**Comment by njacazio** [comment]
Yes but I will work on it only later

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
QC error unrelated. Merging.

**Pull request merged**